Update User Profile Responsive styles

jm90m opened the pull request:
* For medium/small screen sizes: keep username on separate lines, and centered.
* Trello: https://trello.com/c/PWPN1ZVS/255-responsive-user-hero-center-avatar-and-bring-name-in-another-line
* 
![screen shot 2017-10-15 at 2 56 24 pm](https://user-images.githubusercontent.com/5067716/31588072-bafb9cac-b1ba-11e7-921d-5e04469453a7.png)
![responsive-user-page](https://user-images.githubusercontent.com/5067716/31588086-e31d3efc-b1ba-11e7-8372-236be60009eb.gif)

bonustrack commented on src/styles/custom.less:
```diff
@@ -47,3 +47,7 @@
 @layout-zero-trigger-height  : 42px;
+
+// Screen Sizes
+@medium-screen: 990px;
```
We already have existing value for screensize see here: https://github.com/busyorg/busy/blob/702bc7cb698fb8f200a768024cfbc0285258ad13/src/styles/modules/variables.less#L7-L9
I think we the responsive layout should be triggered when size is lower than `768px`. Similar than right sidebar or search in topnav.

jm90m replied:
we can't use these properties, these are set to be `min-width:` rather than `max-width`

Sekhmet commented:
I feel like it would be more logical if `@handle` and user rank was rendered directly below username and above buttons. I'm afraid that it might require some refactoring of UserProfile to make this change though.
What do you think?

jm90m commented:
@bonustrack should we keep username and user rank on same line?
@Sekhmet yeah I agree, should I refactor to move @handle and user rank above the action buttons?

Sekhmet commented:
I think that it should look like this on mobile:
- Avatar
- Username
- Handle
- Rank
- Buttons

On desktop it should still look like it looks now.

bonustrack commented:
I agree too, maybe also these 2 could be on same line:
![image](https://user-images.githubusercontent.com/16245250/31588325-57264dde-b21a-11e7-820c-c06e7295dbe4.png)
What you think?

Sekhmet commented:
Yeah, I think it would make sense to put those buttons on the same line.

jm90m commented:
I like that order @Sekhmet 

@bonustrack so those are 2 action items that do 2 different things, I think its good to keep them in separate lines, since they do different actions. One takes you to a different page while another brings a menu up

bonustrack commented:
@jm90m the button will mostly be "Follow" when not seeing your own profile and the menu [...] appear in down on small screen. IMO it look more sexy if the hero frame has not too much height.

jm90m commented:
@bonustrack ahh I see makes sense then to keep them on same line

bonustrack commented:
Menu [...] could also appear in top right like this:
![image](https://user-images.githubusercontent.com/16245250/31588432-eaaaaac2-b21b-11e7-9f7b-8ff2c4f8a970.png)

jm90m commented:
Just updated to move the menu to top right and move buttons down when using small screen
Also updated order based on what @Sekhmet said.
![screen shot 2017-10-15 at 7 15 59 pm](https://user-images.githubusercontent.com/5067716/31590398-52426894-b1dd-11e7-827c-9e9502ee1d39.png)

bonustrack commented:
- Red and blue line should be same size. Now border in top and in down is not same.
![image](https://user-images.githubusercontent.com/16245250/31608503-c03ba776-b29a-11e7-8f6c-428e66c58eea.png)
- Also there is too much space between name and @handle (see green line)
- I still think we need the @handle and Grade in same line

jm90m commented:
Updated styling to be more aligned @bonustrack 
* also moved handle and rank on same line
<img width="389" alt="screen shot 2017-10-16 at 4 16 30 pm" src="https://user-images.githubusercontent.com/5067716/31633416-51cf3070-b28e-11e7-80c8-db0698c4d8ff.png">
<img width="350" alt="screen shot 2017-10-16 at 4 16 03 pm" src="https://user-images.githubusercontent.com/5067716/31633432-610b75d0-b28e-11e7-8935-90cf6e15e271.png">

<img width="390" alt="screen shot 2017-10-16 at 4 17 28 pm" src="https://user-images.githubusercontent.com/5067716/31633396-4093912a-b28e-11e7-8b8b-bf348fc26921.png">

bonustrack commented:
Almost good. Can you change the breakpoint to `768px` `small`? 
And also can you update the style of the loader too? Now when the account is loading the frame is smaller than what expected in result:
![loading](https://user-images.githubusercontent.com/16245250/31634283-343282d8-b2ed-11e7-910c-fd8135d6beda.gif)

jm90m commented:
@bonustrack alright changed break point and updated loader

<img width="507" alt="screen shot 2017-10-16 at 4 52 58 pm" src="https://user-images.githubusercontent.com/5067716/31635144-ca4d5f04-b293-11e7-8166-eb37e81d01f8.png">
<img width="510" alt="screen shot 2017-10-16 at 4 53 04 pm" src="https://user-images.githubusercontent.com/5067716/31635145-caa49238-b293-11e7-808e-5ad6390a23cb.png">

bonustrack commented:
Idealy it the loader look like this:

![image](https://user-images.githubusercontent.com/16245250/31635374-ab66035e-b2f0-11e7-9302-9dc4905c8d3a.png)

jm90m commented:
@bonustrack ahh i see, should be centered now

![screen shot 2017-10-16 at 5 25 45 pm](https://user-images.githubusercontent.com/5067716/31636073-1f826bc4-b297-11e7-8226-dca73a842430.png)

bonustrack commented:
Can you make the space between element (in red) little bit bigger?
![image](https://user-images.githubusercontent.com/16245250/31638349-4c793a3e-b2fd-11e7-91ec-5065b7456939.png)
Also make the height of the frame bigger so load or not loaded it has the same size, now there is a difference see:
![image](https://user-images.githubusercontent.com/16245250/31638405-9fdc029c-b2fd-11e7-8dda-cdb7a05e185b.png)
For width `768px` in CSS it would be better if we dont hardcode it but use this one https://github.com/busyorg/busy/blob/702bc7cb698fb8f200a768024cfbc0285258ad13/src/styles/modules/variables.less#L8

jm90m commented:
@bonustrack alright made updates based on your suggestions, but we can't use the current variable `@small` on there, since it is `min-width 768px`; for this case we need it to be `max-width: 768px;`

bonustrack commented:
The layout is broken when unlogged, everything appear in same line see:
![image](https://user-images.githubusercontent.com/16245250/31666055-20657872-b375-11e7-8388-d8db63737703.png)

Also it would be nice to keep the button "Follow" even unlogged. Later we would add a popup to login when click on "Follow" while unlogged

jm90m commented:
@bonustrack I updated it to show follow button for unlogged user, that seems to fix layout issue, but also follow button currently does nothing.
* I will create ticket on trello & github to address unlogged follow button issue

bonustrack commented:
@jm90m dont create ticket in Trello anymore, we will use only GitHub. But there is already issues for unlogged actions here:

https://github.com/busyorg/busy/issues/803
https://github.com/busyorg/busy/issues/469
https://github.com/busyorg/busy/issues/315

Sekhmet commented:
I think that we should show Follow button for unlogged user as well as we will ideally show login modal after clicking on it.

Sekhmet commented:
![image](https://user-images.githubusercontent.com/1968722/31675442-f45552f8-b364-11e7-817b-668f6a7d9a49.png)
This is how it looks like at 900px width.

I think that we should add margin-left to avatar here.

jm90m commented:
Added padding for medium / large screen widths, also updated User Header Loader for those screen sizes.

<img width="652" alt="screen shot 2017-10-17 at 1 05 08 pm" src="https://user-images.githubusercontent.com/5067716/31678581-99563354-b33c-11e7-9437-50b80e1466f5.png">

bonustrack commented:
@jm90m avatar should be aligned with the logo when window size is bigger than `998px`
![image](https://user-images.githubusercontent.com/16245250/31680695-460fd750-b39f-11e7-8bdb-ad290aeeaaed.png)

jm90m commented:
@bonustrack removed padding for window bigger than 998px

![screen shot 2017-10-17 at 2 41 08 pm](https://user-images.githubusercontent.com/5067716/31682915-747d91e6-b349-11e7-971c-52f8cf970779.png)

bonustrack commented:
Can we have same padding for window bigger than 998px for loading placeholder too? Then we will be perfect!

jm90m commented:
@bonustrack correct me if I'm wrong but with my latest changes window does have same padding for loading placeholders. My latest changes affected `UserHeader__container` which the loading container is a part of:

![screen shot 2017-10-17 at 3 03 26 pm](https://user-images.githubusercontent.com/5067716/31684043-9f9e7590-b34c-11e7-985d-5fa800c3c161.png)

bonustrack commented:
I mean in left here:
![image](https://user-images.githubusercontent.com/16245250/31684355-aac0fea4-b3a9-11e7-85f0-13e6ccd1d282.png)

Also when loaded it's not aligned to logo
![image](https://user-images.githubusercontent.com/16245250/31684379-c37940d2-b3a9-11e7-962d-88691fcae6d9.png)

jm90m commented:
@bonustrack ahh I see what you mean, looks like the width previously set was not the right length of the alignment, should be updated now!

![screen shot 2017-10-17 at 3 45 45 pm](https://user-images.githubusercontent.com/5067716/31686289-bfbc8e14-b353-11e7-9048-d575c0a58a39.png)
![screen shot 2017-10-17 at 3 52 18 pm](https://user-images.githubusercontent.com/5067716/31686290-bfde0bd4-b353-11e7-94ba-35ae1cb78f38.png)

jm90m commented:
@bonustrack awesome thanks 🙏
